pkg/clusterresourceoverride/internal/handlers/deploy.go

```diff
@@ -117,36 +127,62 @@ func (c *daemonSetHandler) Ensure(ctx *ReconcileRequestContext, cro *autoscaling
 	return
 }
 
-func (c *daemonSetHandler) ApplyToDeploymentObject(context *ReconcileRequestContext, cro *autoscalingv1.ClusterResourceOverride) deploy.Applier {
+func (c *deploymentHandler) ApplyToDeploymentObject(context *ReconcileRequestContext, cro *autoscalingv1.ClusterResourceOverride) deploy.Applier {
 	values := c.asset.Values()
 
 	return func(object metav1.Object) {
-		if len(object.GetAnnotations()) == 0 {
-			object.SetAnnotations(map[string]string{})
+		deployment, ok := object.(*appsv1.Deployment)
+		if !ok {
+			klog.Errorf("expected Deployment, got %T", object)
+			return
+		}
+		if len(deployment.GetAnnotations()) == 0 {
+			deployment.SetAnnotations(map[string]string{})
 		}
 
-		object.GetAnnotations()[values.ConfigurationHashAnnotationKey] = cro.Status.Hash.Configuration
-		object.GetAnnotations()[values.ServingCertHashAnnotationKey] = cro.Status.Hash.ServingCert
+		deployment.GetAnnotations()[values.ConfigurationHashAnnotationKey] = cro.Status.Hash.Configuration
+		deployment.GetAnnotations()[values.ServingCertHashAnnotationKey] = cro.Status.Hash.ServingCert
+
+		// Override replica count, if specified in the CRD
+		if cro.Spec.DeploymentOverrides.Replicas != nil {
+			deployment.Spec.Replicas = cro.Spec.DeploymentOverrides.Replicas
+		}
 
 		context.ControllerSetter().Set(object, cro)
 	}
 }
 
-func (c *daemonSetHandler) ApplyToToPodTemplate(context *ReconcileRequestContext, cro *autoscalingv1.ClusterResourceOverride) deploy.Applier {
+func (c *deploymentHandler) ApplyToToPodTemplate(context *ReconcileRequestContext, cro *autoscalingv1.ClusterResourceOverride) deploy.Applier {
 	values := c.asset.Values()
 
 	return func(object metav1.Object) {
-		if len(object.GetAnnotations()) == 0 {
-			object.SetAnnotations(map[string]string{})
+		podTemplateSpec, ok := object.(*corev1.PodTemplateSpec)
+		if !ok {
+			klog.Errorf("expected PodTemplateSpec, got %T", object)
+			return
+		}
+
+		if len(podTemplateSpec.GetAnnotations()) == 0 {
+			podTemplateSpec.SetAnnotations(map[string]string{})
 		}
 
-		object.GetAnnotations()[values.OwnerAnnotationKey] = cro.Name
-		object.GetAnnotations()[values.ConfigurationHashAnnotationKey] = cro.Status.Hash.Configuration
-		object.GetAnnotations()[values.ServingCertHashAnnotationKey] = cro.Status.Hash.ServingCert
+		podTemplateSpec.GetAnnotations()[values.OwnerAnnotationKey] = cro.Name
+		podTemplateSpec.GetAnnotations()[values.ConfigurationHashAnnotationKey] = cro.Status.Hash.Configuration
+		podTemplateSpec.GetAnnotations()[values.ServingCertHashAnnotationKey] = cro.Status.Hash.ServingCert
+
+		// Replaces nodeSelector, if specified in the CRD
```

Comment:
This should probably be "in the CR" instead of "in the CRD", but anybody reading the comment would know that's what you meant, so only fix it if you end up making other changes prior to merge (or don't worry about it even then).